src/lib/note.ts

```diff
@@ -1,17 +1,22 @@
-function newNote(currentNotes: NoteState[]): NoteState {
+// 新しい空のメモを返す
+function newNote(currentNotes: Note[]): Note {
   const maxId = currentNotes.reduce((a, n) => (a < n.id ? n.id : a), 0);
-  return { id: maxId + 1 };
+  return { id: maxId + 1, x: null };
 }
 
-function appendNewNote(setNotes: SetNotesFunc): void {
-  setNotes(prev => [...prev, newNote(prev)]);
+// TODO: algorithm
+// id に従属するメモの色を返す
+function noteColor(id: number): string {
+  return id % 2 === 0 ? "blue" : "red";
 }
 
-function insertNewNoteBefore(setNotes: SetNotesFunc, id: number): void {
-  setNotes(prev => {
-    const i = prev.findIndex(n => n.id === id);
-    return [...prev.slice(0, i), newNote(prev), ...prev.slice(i, prev.length)];
-  });
+// x 順に並べたときのメモの番号を返す
+function noteNumber(notes: Note[], id: number): number | null {
+  const i = notes
+    .filter(n => n.x !== null)
+    .sort((a, b) => (a.x !== null && b.x !== null ? a.x - b.x : 0))
```

Comment:
TypeScriptコンパイラの型推論が弱いのが問題なので、`!` で強制的に `null` を外してもいいと思う。
```suggestion
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    .sort((a, b) => (a.x! - b.x!))
```